src/librustc_codegen_ssa/mir/block.rs

```diff
@@ -340,6 +362,7 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                  FnType::of_instance(&bx, &drop_fn))
             }
         };
+        bx.sideeffect();
```

Comment:
Would it be possible to insert this on function entry, rather than on every call?